lib/nexus/nexus_plot.cpp

```diff
@@ -39,6 +41,22 @@ const std::string NEXUS_PLOT_TYPE_HEADER = "PLOT  BIN ";
 
 namespace {
 
+float int32_t_bits_to_float(int32_t i) {
+    int32_t sign_bit  = (i & 0x80000000L) >> 31;
+    float   sign      = sign_bit ? -1.0 : 1.0;
+    int32_t exp_bits  = (i & 0x7F800000L) >> 23;
+    int32_t exp       = exp_bits ? exp_bits - 0x0000007FL : -126L;
+    int32_t frac_bits = i & 0x007FFFFFL;
+    double c = (exp_bits ? 1.0 : 0.0) + frac_bits / double(0x007FFFFFL);
+
+    if (exp_bits == 0x000000FFL)
+        return frac_bits ? NAN : (sign_bit ? -INFINITY : INFINITY);
+    else if (frac_bits | exp_bits)
+        return float(sign * c * std::pow(2, exp));
+    else
+        return 0.0f;
+}
```

Comment:
maybe I'm missing something here, but wouldn't it be sufficient to `return static_cast<float>(i)`